packages/SwingSet/src/kernel/vatManager/deliver.js

```diff
@@ -81,29 +81,26 @@ export function makeDeliver(tools, dispatch) {
     );
   }
 
-  async function deliverOneNotification(vpid, vp) {
-    const errmsg = `vat[${vatID}].promise[${vpid}] ${vp.state} failed`;
-    switch (vp.state) {
-      case 'fulfilledToPresence':
-        return doProcess(['notifyFulfillToPresence', vpid, vp.slot], errmsg);
-      case 'redirected':
-        // TODO unimplemented
-        throw new Error('not implemented yet');
-      case 'fulfilledToData':
-        return doProcess(['notifyFulfillToData', vpid, vp.data], errmsg);
-      case 'rejected':
-        return doProcess(['notifyReject', vpid, vp.data], errmsg);
-      default:
-        throw Error(`unknown promise state '${vp.state}'`);
+  async function deliverOneNotification(primaryVpid, resolutions) {
+    for (const vpid of Object.keys(resolutions)) {
+      // XXX return inside loop is wrong once `resolutions` has more than 1 element
+      const vp = resolutions[vpid];
+      const errmsg = `vat[${vatID}].promise[${vpid}] ${vp.rejected} failed`;
+      return doProcess(['notify', vpid, vp.rejected, vp.data], errmsg);
```

Comment:
Looks like we'll have a "step 1.5", to change liveslots to accept a `dispatch.notify` which takes a multi-promise resolution set, since liveslots really needs to hear about the whole set in a single batch, rather than getting a series of `dispatch.notify` calls.